business/istio_validations.go

```diff
@@ -262,6 +265,14 @@ func (in *IstioValidationsService) fetchNonLocalmTLSConfigs(mtlsDetails *kuberne
 		return
 	}
 
+	meshPolicies, err := in.k8s.GetMeshPolicies(config.Get().IstioNamespace)
```

Comment:
This blocks the next fetch. Is there a requirement to block processing while fetching this or should it happen in a goroutine?

Reply:
No, there isn't. However, this call is within a goroutine already. Plus, there is one more call using goroutines. So I prevented myself to add one more level of concurrency to avoid scenarios where there were more work to do for managing concurrency than the task itself.

Reply:
I updated it with adding only concurrency into that first call.